cudapoa/src/cudapoa_nw_banded.cuh

```diff
@@ -556,6 +556,122 @@ __device__ __forceinline__
     return aligned_nodes;
 }
 
+// global kernel used in testing, hence uses int16_t for SizeT and ScoreT,
+// may need to change if test inputs change to long reads
+template <typename SizeT>
+__global__ void runNeedlemanWunschBandedKernel(uint8_t* nodes,
+                                               SizeT* graph,
+                                               SizeT* node_id_to_pos,
+                                               int32_t graph_count,
+                                               uint16_t* incoming_edge_count,
+                                               SizeT* incoming_edges,
+                                               uint16_t* outgoing_edge_count,
+                                               uint8_t* read,
+                                               int32_t read_length,
+                                               int16_t* scores,
+                                               int32_t scores_width,
+                                               int32_t max_nodes_per_graph,
+                                               SizeT* alignment_graph,
+                                               SizeT* alignment_read,
+                                               int32_t band_width,
+                                               int32_t gap_score,
+                                               int32_t mismatch_score,
+                                               int32_t match_score,
+                                               SizeT* aligned_nodes,
+                                               bool adaptive)
+{
+    float banded_buffer_size = static_cast<float>(max_nodes_per_graph) * static_cast<float>(scores_width);
+
+    if (adaptive)
```

Comment:
since this is only for test case use with a small dataset, it does not really impact on performance. Changed it not to pass `bool adaptive` to device kernel.